zebedee-reader/src/main/java/com/github/onsdigital/zebedee/search/api/endpoint/ReIndex.java

```diff
@@ -1,103 +0,0 @@
-
-    @POST
-    public Object reIndex(@Context HttpServletRequest request, @Context HttpServletResponse response) throws IOException {
-
-        try {
-            String key = request.getParameter("key");
-            if (Password.verify(key, REINDEX_KEY_HASH)) {
-                boolean reindexAll = "1".equals(request.getParameter("all"));
-                if (reindexAll) {
-                    Indexer.getInstance().reload();
-                } else {
-                    String uri = request.getParameter("uri");
-                    if (StringUtils.isEmpty(uri)) {
-                        return "Please specify uri of the content to reindex";
-                    } else {
-                        Indexer.getInstance().reloadContent(uri);
-                    }
-                }
-                response.setStatus(HttpStatus.OK_200);
-                return "Elasticsearch: indexing complete";
-            } else {
-                response.setStatus(HttpStatus.UNAUTHORIZED_401);
-                return "Wrong key, make sure you pass in the right key";
-            }
-        } catch (IndexInProgressException e) {
-            response.setStatus(HttpStatus.CONFLICT_409);
-            return "Indexing already in progress.";
-        } catch (IndexingException e) {
-            e.printStackTrace();
-            response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR_500);
-            return "Indexing failed! %s" + e.getMessage();
-        }
-    }
-
-    @DELETE
-    public Object deleteIndex(@Context HttpServletRequest request, @Context HttpServletResponse response) throws IOException {
-
-        try {
-            String key = request.getParameter("key");
-            if (Password.verify(key, REINDEX_KEY_HASH)) {
-
-                String uri = request.getParameter("uri");
-                String pageType = request.getParameter("pageType");
-                if (StringUtils.isEmpty(uri)) {
-                    return "Please specify uri of the content to delete";
-                } else {
-                    Indexer.getInstance().deleteContentIndex(pageType, uri);
-                }
-
-                response.setStatus(HttpStatus.OK_200);
-                return "Elasticsearch: indexing complete";
-            } else {
-                response.setStatus(HttpStatus.UNAUTHORIZED_401);
-                return "Wrong key, make sure you pass in the right key";
-            }
-        } catch (IndexInProgressException e) {
-            response.setStatus(HttpStatus.CONFLICT_409);
-            return "Indexing already in progress.";
-        } catch (IndexingException e) {
-            e.printStackTrace();
-            response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR_500);
-            return "Indexing failed! %s" + e.getMessage();
-        }
-    }
-
-    /**
-     * Generates new re-indexing key/hash values.
-     *
-     * @param args Not used.
```

Comment:
Haha, this is duplicated code from the bit I've said to remove in the other comment. Nothing to do here, just sharing the humour! 😆